src/base/base.c

```diff
@@ -602,6 +604,11 @@ void xdebug_base_rinit()
 	}
 
 	XG_BASE(stack) = xdebug_llist_alloc(function_stack_entry_dtor);
+	if (XINI_BASE(max_nesting_level) != -1 && XINI_BASE(max_nesting_level) <= 256) {
+		XG_BASE(stackPool) = xdmalloc (sizeof (function_stack_entry) * XINI_BASE(max_nesting_level));
+	} else {
+		XG_BASE(stackPool) = NULL;
```

Comment:
I would prefer setting this default in RINIT and / init_globals instead, instead, so you don't need the else case here.